src/interface/params.hpp

```diff
@@ -43,10 +43,11 @@ class Params {
   ///
   /// Throws an error if the key is already in use
   template <typename T>
-  void Add(const std::string &key, T value) {
+  void Add(const std::string &key, T value, bool make_mutable = false) {
```

Comment:
Nitpicking: I suggest `is_mutable` (here and below) as I see this more like a property of the parameter rather than as an a posteriori action.

Reply:
done.